server/src/api/listings.ts

```diff
@@ -25,19 +25,42 @@ import {listingService} from '../services';
 
 const listingRouter = Router();
 
+/**
+ * Handles the get request for retrieving all listings that satisfy the query
+ * parameters.
+ * Querying by ids is a restricted query, once the id field is specified, requester
+ * would not be allowed to filter by any other query params.
+ * Otherwise, it would be a Bad Request.
+ */
 listingRouter.get(
   '/',
   async (req: Request, res: Response, next: NextFunction) => {
     try {
+      const {ids: idsStr} = req.query;
+      if (idsStr !== undefined) {
+        if (Object.keys(req.query).length > 1) {
+          res.sendStatus(400);
+          return;
+        }
+
+        const ids = (idsStr as string).split(',').map(idStr => {
```

Comment:
should we allow empty ids?

Reply:
@ifhuang Good question, I think I would make it a 400.